src/util/dependencyGraph.ts

```diff
@@ -54,30 +49,60 @@ export function runGraph(
   readFraction: number,
   framework: ReactiveFramework
 ): number {
-  const rand = pseudoRandom();
+  const rand = new Random("seed");
   const { sources, layers } = graph;
   const leaves = layers[layers.length - 1];
   const skipCount = Math.round(leaves.length * (1 - readFraction));
   const readLeaves = removeElems(leaves, skipCount, rand);
+  const frameworkName = framework.name.toLowerCase();
+  // const start = Date.now();
+  let sum = 0;
+
+  if (frameworkName === "s-js" || frameworkName === "solidjs") {
+    // [S.js freeze](https://github.com/adamhaile/S#sdatavalue) doesn't allow different values to be set during a single batch, so special case it.
+    for (let i = 0; i < iterations; i++) {
+      framework.withBatch(() => {
+        const sourceDex = i % sources.length;
+        sources[sourceDex].write(i + sourceDex);
+      });
+
+      for (const leaf of readLeaves) {
+        leaf.read();
+      }
+    }
 
-  for (let i = 0; i < iterations; i++) {
+    sum = readLeaves.reduce((total, leaf) => leaf.read() + total, 0);
+  } else {
```

Comment:
This allows for an algorithm that defers all changes until after a batch is finished to only walk the graph once instead of `iterations` times (especially because we have no verification on what leaf.read returns.

Would it be possible to have all frameworks use the same path? (and in a future change I'll add assertions to ensure that every framework's leaf.read is actually getting the right result)